src/ImageRunModal.jsx

```diff
@@ -342,7 +342,24 @@ export class ImageRunModal extends React.Component {
         });
     };
 
+    buildFilterRegex(searchText) {
+        // Strip out all non-allowed container image characters when filtering.
+        let regexString = searchText.replace(/[^/\w_.:-]/g, "");
+        // Strip image registry option if set for comparing results for docker.io searching for docker.io/fedora
+        // returns docker.io/$username/fedora for example.
+        if (regexString.includes('/')) {
+            regexString = searchText.replace(searchText.split('/')[0], '');
+        }
+
+        return new RegExp(regexString, 'i');
+    }
+
     onSearchTriggered = value => {
+        const imageExistsLocally = (searchText, localImages) => {
+            const regex = this.buildFilterRegex(searchText);
+            return localImages.some(localImage => localImage.Name.search(regex) !== -1);
```

Comment:
`buildFilterRegex` strips off the first `/` component (the registry name). This *may* be needed for the docker.io search API (see my question above), but it's wrong here. If I search for "docker.io/fedora" and have a local "quay.io/mpitt-fedora/test:img" image, this should *not* match. This definitively needs to include the registry here. So maybe make that an option of that function?

Reply:
True, thanks for catching this